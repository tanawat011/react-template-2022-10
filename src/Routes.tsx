import { BrowserRouter, Routes, Route } from 'react-router-dom'

import { AuthenticationContainer } from 'containers/Authentication'
import { BackofficeContainer } from 'containers/Backoffice'
import { ErrorContainer } from 'containers/Error'
import { ChangePassword, ForgotPassword, Login } from 'pages/Authentication'
import { Error401, Error403, Error404, Error500, Error502, Error503, Error504 } from 'pages/Error'
import { Example } from 'pages/Example'
import { HuesAndCues } from 'pages/Game/HuesAndCues'
import { Home } from 'pages/Home'
import { ReadCsv } from 'pages/ReadCsv'
import { Todo } from 'pages/Todo/TodoDetail'

type Path = {
  AUTH: {
    ROOT: string
    LOGIN: string
    CHANGE_PASSWORD: string
    FORGOT_PASSWORD: string
  }
  BACKOFFICE: {
    ROOT: string
    HOME: string
    TODO: string
    ABOUT: string
    GAME: {
      ROOT: string
      HUES_AND_CUES: {
        ROOT: string
        SET_DISPLAY_NAME: string
        SETUP_ROOM: string
      }
    }
    READ_CSV: string
  }
  ERROR: {
    ROOT: string
    401: string
    403: string
    404: string
    500: string
    502: string
    503: string
    504: string
  }
}

export const PATH: Path = {
  AUTH: {
    ROOT: 'auth',
    LOGIN: 'login',
    CHANGE_PASSWORD: 'change-password',
    FORGOT_PASSWORD: 'forgot-password',
  },
  BACKOFFICE: {
    ROOT: '/',
    HOME: 'home',
    TODO: 'todo',
    ABOUT: 'about',
    GAME: {
      ROOT: 'game',
      HUES_AND_CUES: {
        ROOT: 'hues-and-cues',
        SET_DISPLAY_NAME: 'set-display-name',
        SETUP_ROOM: 'setup-room',
      },
    },
    READ_CSV: 'read-csv',
  },
  ERROR: {
    ROOT: '',
    401: '401',
    403: '403',
    404: '404',
    500: '500',
    502: '502',
    503: '503',
    504: '504',
  },
}

export const AppRoutes = () => {
  const { AUTH, BACKOFFICE, ERROR } = PATH
  const { ROOT: GAME, HUES_AND_CUES } = BACKOFFICE.GAME

  return (
    <BrowserRouter>
      <Routes>
        <Route path={AUTH.ROOT} element={<AuthenticationContainer />}>
          <Route index element={<Login />} />
          <Route path={AUTH.LOGIN} element={<Login />} />
          <Route path={AUTH.CHANGE_PASSWORD} element={<ChangePassword />} />
          <Route path={AUTH.FORGOT_PASSWORD} element={<ForgotPassword />} />
        </Route>

        <Route path={BACKOFFICE.ROOT} element={<BackofficeContainer />}>
          <Route index element={<Home />} />
          <Route path={BACKOFFICE.HOME} element={<Home />} />
          <Route path={BACKOFFICE.TODO} element={<Todo />} />
          <Route path={BACKOFFICE.ABOUT} element={<Example />} />
          <Route path={GAME}>
            <Route path={HUES_AND_CUES.ROOT}>
              <Route path={':roomId'} element={<HuesAndCues />} />
            </Route>
          </Route>
          <Route path={BACKOFFICE.READ_CSV} element={<ReadCsv />} />
        </Route>

        <Route path={ERROR.ROOT} element={<ErrorContainer />}>
          <Route index element={<Error404 />} />
          <Route path={ERROR[401]} element={<Error401 />} />
          <Route path={ERROR[403]} element={<Error403 />} />
          <Route path={ERROR[404]} element={<Error404 />} />
          <Route path={ERROR[500]} element={<Error500 />} />
          <Route path={ERROR[502]} element={<Error502 />} />
          <Route path={ERROR[503]} element={<Error503 />} />
          <Route path={ERROR[504]} element={<Error504 />} />
        </Route>

        <Route path='*' element={<Error404 />} />
      </Routes>
    </BrowserRouter>
  )
}
