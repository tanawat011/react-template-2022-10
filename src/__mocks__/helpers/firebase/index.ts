export const deleteCollection = jest.fn()
export const deleteDocument = jest.fn()
export const getDocument = jest.fn()
export const getAllDocuments = jest.fn()
export const setDocument = jest.fn()
export const subscribeCollection = jest.fn()
export const updateDocument = jest.fn()
