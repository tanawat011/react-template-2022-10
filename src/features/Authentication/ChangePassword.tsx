export const ChangePassword = () => {
  return <>ChangePassword</>
}
