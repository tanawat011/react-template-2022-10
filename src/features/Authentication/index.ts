export * from './ChangePassword'
export * from './ForgotPassword'
export * from './Login'
