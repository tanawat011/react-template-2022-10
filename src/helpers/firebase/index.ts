export * from './deleteCollection'
export * from './deleteDocument'
export * from './getAllDocuments'
export * from './getDocument'
export * from './setDocument'
export * from './subscribeCollection'
export * from './updateDocument'
