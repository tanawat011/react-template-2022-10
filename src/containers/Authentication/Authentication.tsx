import { useEffect } from 'react'

import { Outlet, useLocation, useNavigate } from 'react-router-dom'
import tw from 'twin.macro'

import { PATH } from 'Routes'
import { TwCol, TwRow } from 'components/Common'

const TwLoginContainer = tw(TwRow)`w-full h-screen bg-slate-300`
const TwLoginLeftSection = tw(TwCol)`w-1/2 items-center justify-center bg-slate-400`
const TwLoginRightSection = tw(TwCol)`w-1/2 items-center justify-center bg-slate-500`

export const AuthenticationContainer: React.FC = () => {
  const location = useLocation()
  const navigate = useNavigate()

  useEffect(() => {
    handleRedirectToLogin()
  }, [])

  const handleRedirectToLogin = () => {
    const pathAuth = `/${PATH.AUTH.ROOT}`

    if (location.pathname === pathAuth) {
      navigate(`${pathAuth}/${PATH.AUTH.LOGIN}`)
    }
  }

  return (
    <TwLoginContainer>
      <TwLoginLeftSection>Left Section</TwLoginLeftSection>

      <TwLoginRightSection>
        <Outlet />
      </TwLoginRightSection>
    </TwLoginContainer>
  )
}
