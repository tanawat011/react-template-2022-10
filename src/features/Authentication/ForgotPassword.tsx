import { Link } from 'react-router-dom'

import { PATH } from 'Routes'

export const ForgotPassword = () => {
  return (
    <>
      <p>ForgotPassword</p>
      <Link to={`../${PATH.AUTH.LOGIN}`}>Login</Link>
    </>
  )
}
