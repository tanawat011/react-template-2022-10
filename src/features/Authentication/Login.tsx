import { PATH } from 'Routes'
import { Button } from 'components/Button'

export const Login = () => {
  return (
    <>
      <div id='login-form'>
        <h1>Login</h1>

        <form className='flex flex-col'>
          <label id='username'>Username</label>
          <input id='username' />
          <label id='password'>Password</label>
          <input id='password' />

          <div className='flex'>
            <Button link to={`/${PATH.BACKOFFICE.HOME}`}>
              Login
            </Button>
            <Button link to={`../${PATH.AUTH.FORGOT_PASSWORD}`}>
              Forgot Password
            </Button>
          </div>
        </form>
      </div>
    </>
  )
}
