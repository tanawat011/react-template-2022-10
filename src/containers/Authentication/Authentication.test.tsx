import { PATH } from 'Routes'
import { renderWithProviders } from 'helpers/test'

import { AuthenticationContainer } from './Authentication'

describe('<AuthenticationContainer />', () => {
  test('should render', () => {
    const { container } = renderWithProviders(<AuthenticationContainer />, {
      router: { route: `/${PATH.AUTH._PATH}` },
    })
    expect(container).toBeInTheDocument()
  })

  test('match snapshot', () => {
    const { asFragment } = renderWithProviders(<AuthenticationContainer />)
    expect(asFragment()).toMatchSnapshot()
  })
})
