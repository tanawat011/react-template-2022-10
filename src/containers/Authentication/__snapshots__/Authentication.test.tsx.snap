// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthenticationContainer /> match snapshot 1`] = `
<DocumentFragment>
  <p>
    AuthenticationContainer
  </p>
</DocumentFragment>
`;
